platform/darwin/src/MGLGeometry.h

```diff
@@ -181,4 +181,7 @@ NS_INLINE CLLocationDegrees MGLDegreesFromRadians(CGFloat radians) {
     return radians * 180 / M_PI;
 }
 
+/** Returns mercator projection of a WGS84 coordinate at the specified zoom level. */
+CLLocationCoordinate2D MGLProjectCoordinate(CLLocationCoordinate2D coordinate, double zoomLevel);
```

Comment:
I keep reading “project” as a noun. Since this function doesn’t mutate its parameters, let’s give it a name like `MGLMapPointForCoordinate()`. That would be reminiscent with [`MKMapPointForCoordinate()`](https://developer.apple.com/documentation/mapkit/1452514-mkmappointforcoordinate). [In MapKit](https://developer.apple.com/library/content/documentation/UserExperience/Conceptual/LocationAwarenessPG/MapKit/MapKit.html#//apple_ref/doc/uid/TP40009497-CH3-SW6), a “map point” is a projected point in the Web Mercator coordinate system.

`CLLocationCoordinate2D` is always by definition a WGS 84 coordinate. Use `CGPoint` or a new `MGLMapPoint` struct to represent the projected coordinate. A new type would be ideal, because our concept of a projected coordinate appears to be dependent on the zoom level, unlike `MKMapPoint`, and `CGPoint` doesn’t store a third value.